pkg/test/clusterpool/clusterpool.go

```diff
@@ -94,6 +94,12 @@ func WithSize(size int) Option {
 	}
 }
 
+func WithLabels(labels map[string]string) Option {
```

Comment:
This is confusing since it is not clear from the name of the function whether it is setting `.metadata.labels` or `.spec.labels`.